module/Education/src/Education/Controller/AdminController.php

```diff
@@ -12,6 +12,31 @@ public function indexAction()
     {
     }
 
+    public function addCourseAction()
+    {
+        $service = $this->getExamService();
+        $request = $this->getRequest();
+
+        if ($request->isPost()) {
+            // try uploading
+            if ($service->AddCourse($request->getPost())) {
+                return new ViewModel([
+                    'form' => $service->getAddCourseForm(),
+                    'success' => true
+                ]);
+            }
+            $this->getResponse()->setStatusCode(500);
```

Comment:
500 is definitely not the correct response code when validation has failed. 500 means that there was a problem on our end, while failing validation is a problem by the user. If you want to set a different code, look at 4xx codes. Though just returning 200 with validation errors is good enough.